flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -473,8 +473,13 @@ class JobManager(
 
       val jobInfo = new JobInfo(client, listeningBehaviour, System.currentTimeMillis(),
         jobGraph.getSessionTimeout)
-
-      submitJob(jobGraph, jobInfo)
+      val jobID = jobGraph.getJobID
+      if (!currentJobs.contains(jobID)) {
```

Comment:
Yea, I noticed this as well. I am going to sit on this for a bit until I get a reply back from the dev email thread. They are re-writing the JobManager so fixing that and adding guards around job submission might be best included in that work instead of implementing it now.